demos/CMakeLists.txt

```diff
@@ -35,11 +35,41 @@ function(check_aws_credentials demo_name)
     endif()
 endfunction()
 
+if(DOWNLOAD_CERTS)
+    # Download the Amazon Root CA certificate.
+    message( "Downloading the Amazon Root CA certificate..." )
+    file(MAKE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/certificates)
+    execute_process(
+        COMMAND curl --url https://www.amazontrust.com/repository/AmazonRootCA1.pem
+        -o ${CMAKE_CURRENT_LIST_DIR}/certificates/AmazonRootCA1.crt
+    )
+endif()
+
+# Copy the certificates and client keys to the build directory.
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/certificates"
+     DESTINATION "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+if(BUILD_TESTS)
+    file(COPY "${CMAKE_CURRENT_LIST_DIR}/certificates"
+         DESTINATION "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/tests")
+endif()
+
+# Set prefix to PWD if any path flags are relative
+if(DEFINED ENV{PWD})
+    if(NOT ROOT_CA_CERT_PATH MATCHES "/$")
```

Comment:
What if these variables are not passed in the CMake command in which case they would (probably) have empty values? 
Would these generate invalid variables for the credential variables in that case?

Reply:
I tested the `mqtt_demo_basic_tls` locally with not providing the `ROOT_CA_CERT_PATH` with the following command:
`cmake .. -DBUILD_TESTS="ON" -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS='-Wextra -Wall -O0 -ggdb'`

The build logs show that the absolute path logic is causing the `ROOT_CA_CERT_PATH` CMake variable to be defined even those it wasn't passed by me, and thus, it is building the demo target with an incorrect value (of `/home/ubuntu/Repos/aws-iot-device-sdk-embedded-C/build/` as can be seen in the logs)
```
╭─ubuntu@ip-172-31-24-209 ~/Repos/aws-iot-device-sdk-embedded-C/build  ‹development*› 
╰─➤  ./bin/mqtt_demo_basic_tls         
[INFO] [DEMO] [mqtt_demo_basic_tls.c:400] Establishing a TLS session to test.mosquitto.org:8883.
[DEBUG] [Sockets] [sockets_posix.c:170] Performing DNS lookup: Host=test.mosquitto.org.
[DEBUG] [Sockets] [sockets_posix.c:210] Attempting to connect to server: Host=test.mosquitto.org, IP address=5.196.95.208.
[DEBUG] [Sockets] [sockets_posix.c:225] Connected to IP address: 5.196.95.208.
[DEBUG] [Sockets] [sockets_posix.c:256] Established TCP connection: Server=test.mosquitto.org.

[DEBUG] [Transport_OpenSSL_Sockets] [openssl_posix.c:144] Attempting to open Root CA certificate: Path=/home/ubuntu/Repos/aws-iot-device-sdk-embedded-C/build/.
[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:195] PEM_read_X509 failed to parse root CA.
[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:474] Setting up credentials failed.
```

Reply:
The fix is simple by just updating each of the nested if conditions to check if their respective CMake variables are defined